### ISSUE TITLE
Generating an EML from an Email object with nested attachments, when created from an Outlook MSG file

### ISSUE THREAD

**Issue by atmcq** [bbottema/simple-java-mail#315]
Following #298 I can access nested Outlook attachments within Email objects that have been created by EmailConverter.outlookMsgToEmail().

However if I subsequently use EmailConverter.emailToEML() on the Email object created in this way, the nested Outlook attachment is replaced with the serialised object as an SJM file, e.g. "attachment 1 as nested Outlook message (converted).sjm".

It looks like it needs to be deserialised and converted back to a .MSG or .EML attachment as part of emailToEML(). 

```java
public static void main(String[] args) throws IOException {
    viewMsg();	
}
	
public static void viewMsg() throws IOException {
    Email em1 = EmailConverter.outlookMsgToEmail(new File("Email2.msg"));
    String emlStr = EmailConverter.emailToEML(em1);
    FileOutputStream outputStream = new FileOutputStream("NewEml.eml");
    byte[] strToBytes = emlStr.getBytes();
    outputStream.write(strToBytes);
    outputStream.close();
}
```

[Examples.zip](https://github.com/bbottema/simple-java-mail/files/6442593/Examples.zip)

**Comment by atmcq** [comment]
>> It looks like it needs to be deserialised and converted back to a .MSG or .EML attachment as part of emailToEML().

My thinking here was that if I am using emailToEml() then the user may need to access the nested attachment in the same mail client that needs .eml and therefore converting the nested attachment to .eml makes sense.

**Comment by bbottema** [comment]
So with the latest change, that .sjm file can be accessed as follows:

```
	// email.getAttachments().get(0).getName() == "attachment 0 as nested Outlook message (converted).sjm"
	InputStream sjmInputstream = email.getAttachments().get(0).getDataSourceInputStream();
	Email nestedOutlookMessage = SerializationUtil.deserialize(sjmInputstream);
```
However, I didn't realize I could just serialize to an EML string instead, which agree with you is the better solution, because this removes the need for Kryo to handle datasource serialization. Also the file is self-describing, as people would generally recognize .eml, but certainly not .sjm.

I'll perform the changes when I have some time. Thanks for thinking along with me on this.

**Comment by LaurentFl** [comment]
Great, thanks! I also agree for EML, this makes sense. Waiting this :-)
In Outlook .msg attachment filename is taken from attachement .msg subject. Maybe the same can be done:
`builder.withAttachment(email.getSubject() != null ? email.getSubject() : "attachment " + i + " as nested Outlook message (converted).sjm", new ByteArrayDataSource(MiscUtil.serialize(EmailConverter.emailToMimeMessage(email).getInputStream()), "application/octet-stream"));`
But it can produce errors or be dangerous if specials characters are present in subject...
(And not sur of this part of code, just for example: `new ByteArrayDataSource(MiscUtil.serialize(EmailConverter.emailToMimeMessage(email).getInputStream())`)

**Comment by bbottema** [comment]
It's good to be reminded every now and then that this library is actually being used in real life. I know from the download figures that this is the case of course, but without many but reports it's easy to forget.

I've released a fix for this in 6.5.4. Nested attachments are now stored as regular EML attachments and also I started using Outlook's own strategy for naming these nested messages, by simply using their subject as attachment/file name.

Cheers!
